options/options.go

```diff
@@ -329,64 +347,75 @@ func NewTerragruntOptionsForTest(terragruntConfigPath string) (*TerragruntOption
-		StrictInclude:                  terragruntOptions.StrictInclude,
-		RunTerragrunt:                  terragruntOptions.RunTerragrunt,
-		AwsProviderPatchOverrides:      terragruntOptions.AwsProviderPatchOverrides,
-		HclFile:                        terragruntOptions.HclFile,
-		JSONOut:                        terragruntOptions.JSONOut,
-		Check:                          terragruntOptions.Check,
-		CheckDependentModules:          terragruntOptions.CheckDependentModules,
-		FetchDependencyOutputFromState: terragruntOptions.FetchDependencyOutputFromState,
-		UsePartialParseConfigCache:     terragruntOptions.UsePartialParseConfigCache,
-		OutputPrefix:                   terragruntOptions.OutputPrefix,
-		IncludeModulePrefix:            terragruntOptions.IncludeModulePrefix,
+		Logger:                         util.CreateLogEntryWithWriter(opts.ErrWriter, workingDir, opts.LogLevel, opts.Logger.Logger.Hooks),
+		LogLevel:                       opts.LogLevel,
+		ValidateStrict:                 opts.ValidateStrict,
+		Env:                            util.CloneStringMap(opts.Env),
+		Source:                         opts.Source,
+		SourceMap:                      opts.SourceMap,
+		SourceUpdate:                   opts.SourceUpdate,
+		DownloadDir:                    opts.DownloadDir,
+		Debug:                          opts.Debug,
+		OriginalIAMRoleOptions:         opts.OriginalIAMRoleOptions,
+		IAMRoleOptions:                 opts.IAMRoleOptions,
+		IgnoreDependencyErrors:         opts.IgnoreDependencyErrors,
+		IgnoreDependencyOrder:          opts.IgnoreDependencyOrder,
+		IgnoreExternalDependencies:     opts.IgnoreExternalDependencies,
+		IncludeExternalDependencies:    opts.IncludeExternalDependencies,
+		Writer:                         opts.Writer,
+		ErrWriter:                      opts.ErrWriter,
+		MaxFoldersToCheck:              opts.MaxFoldersToCheck,
+		AutoRetry:                      opts.AutoRetry,
+		RetryMaxAttempts:               opts.RetryMaxAttempts,
+		RetrySleepIntervalSec:          opts.RetrySleepIntervalSec,
+		RetryableErrors:                util.CloneStringList(opts.RetryableErrors),
+		ExcludeDirs:                    opts.ExcludeDirs,
+		IncludeDirs:                    opts.IncludeDirs,
+		ModulesThatInclude:             opts.ModulesThatInclude,
+		Parallelism:                    opts.Parallelism,
+		StrictInclude:                  opts.StrictInclude,
+		RunTerragrunt:                  opts.RunTerragrunt,
+		AwsProviderPatchOverrides:      opts.AwsProviderPatchOverrides,
+		HclFile:                        opts.HclFile,
+		JSONOut:                        opts.JSONOut,
+		Check:                          opts.Check,
+		CheckDependentModules:          opts.CheckDependentModules,
+		FetchDependencyOutputFromState: opts.FetchDependencyOutputFromState,
+		UsePartialParseConfigCache:     opts.UsePartialParseConfigCache,
+		OutputPrefix:                   opts.OutputPrefix,
+		IncludeModulePrefix:            opts.IncludeModulePrefix,
```

Comment:
NIT: this is a already a massive refactor (8K lines of code change!), and renames like this add a lot of lines of code for something that is largely unrelated. In the future, please do this sort of change (variable rename) in a separate PR from, say, refactoring how CLI commands/args are handled.

Reply:
Fair point, next time I will not clog the pull request with such changes. Sorry.